compiler/noirc_evaluator/src/ssa/opt/assert_constant.rs

```diff
@@ -112,6 +112,10 @@ fn evaluate_static_assert(
         Ok(false)
     } else {
         let call_stack = function.dfg.get_call_stack(instruction);
-        Err(RuntimeError::StaticAssertFailed { call_stack })
+        if function.dfg.is_constant(arguments[0]) {
+            Err(RuntimeError::StaticAssertFailed { call_stack })
+        } else {
+            Err(RuntimeError::StaticAssertDynamicPredicate { call_stack })
+        }
```

Comment:
I think this'd be easier to read if we removed the `is_constant_true` and just had a `get_constant_bool` or similar, did our `if condition` inside there, and for the else of the outer `get_constant_bool` case put the dynamic error.

Not worth blocking over though.